source/codegen/templates/service.h.mako

```diff
@@ -18,6 +18,12 @@ if len(config["custom_types"]) > 0:
 resource_handle_type = config.get("resource_handle_type", "ViSession")
 resource_repository_type = f"nidevice_grpc::SessionResourceRepository<{resource_handle_type}>"
 resource_repository_ptr = f"std::shared_ptr<{resource_repository_type}>"
+
+async_functions = service_helpers.get_async_functions(functions)
+has_async_functions = any(async_functions)
+base_class_name = f"{service_class_prefix}::Service"
+for async_function in async_functions.keys():
+  base_class_name = f"{service_class_prefix}::WithAsyncMethod_{async_function}<{base_class_name}>"
```

Comment:
I couldn't find the `WithAsyncMethod` documentation but is it fine to just use the last `async_function` key here for the `base_class_name`? I know right now we only have `RegisterDoneEvent` but are we planning on adding any more async methods and will changing this be fine? Does it just have to be one of the async_function names?

Reply:
The idea is that they're supposed to nest, so each time through the loop, we will wrap the previous result.

So, it should come out as 

`class DAQService: WithAsyncMethod_RegisterDoneEvent<WithAsyncMethod_RegisterEveryNSamplesEvent<Service>>

Here are [some examples](https://grep.app/search?current=3&q=WithAsyncMethod_%5BA-Za-z%5D%2B%3C&regexp=true). But there's definitely not much documentation on it.

Reply:
Oh, I just misread it. That makes sense.